### DISCUSSION TITLE
Stochstic term in Allen-Cahn equation

### DISCUSSION THREAD

**Discussion by mdh23rct** [idaholab/moose#28429] [Q&A Modules: Phase field]
Dear MOOSE experts,
Would you be kind to let me know how to add a stochastic term in the Allen-Cahn equation in MOOSE?
![stochstic](https://github.com/user-attachments/assets/fe5862c5-6836-4ae6-8719-056984a6a74d)
I have attached an image of the equation here. The last term on the right-hand side of the equation stands for the stochastic term. 

I would appreciate if you could manage time to reply.

Thanks, and best regards.

**Comment by lynnmunday** [comment]
Would some kind of initial condition for that term work seeded with a random distribution like this:
``` 
[AuxVariables]
  [./strength]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[ICs]
  [./strength]
    type = VolumeWeightedWeibull
    variable = strength
    reference_volume = 1e-7
    weibull_modulus = 12.0
    median = 130e6
  [../]
[]
```

**Reply by mdh23rct** [reply]
Dear Dr. @lynnmunday,
Thank you for your advice. But how to add the stochastic term with the Allen-Chan equation? Do I need any kernel for that?

Thanks, and best regards.

**Reply by lynnmunday** [reply]
I'm not familiar with this equation but if you just need to seed your domain with a random material property, this is the way to do that and then you can read that material property into your model as is done in the above example.

**Reply by zahidhasan83** [reply]
@lynnmunday
Dear Dr. Lynn,
I need to add a random number (R3), as shown in this equation:
![random number](https://github.com/user-attachments/assets/0b09b217-3fb8-4217-8ded-6e1cb907cce4)
Would you be kind to advise me how to do it in MOOSE?

Thanks, and best regards.

**Reply by lynnmunday** [reply]
I have no idea what this equation is doing.   What I have above is something that will seed an auxvariable with a random number over some distribution.  It looks like you could random multiply this auxvariable by your material.

**Reply by lynnmunday** [reply]
@chaibhave Will you look at this.

**Reply by Nuwanted** [reply]
May I ask if you have solved this problem? I also encountered the same issue
